src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -112,25 +156,17 @@ export class RuntimeLanguageContext {
         T extends codewhispererClient.ListRecommendationsRequest | codewhispererClient.GenerateRecommendationsRequest
     >(request: T): T {
         const fileContext = request.fileContext
-        const childLanguage = request.fileContext.programmingLanguage
-        let parentLanguage: codewhispererClient.ProgrammingLanguage
-        switch (childLanguage.languageName) {
-            case 'tsx':
-                parentLanguage = { languageName: CodeWhispererConstants.typescript }
-                break
-            case 'jsx':
-                parentLanguage = { languageName: CodeWhispererConstants.javascript }
-                break
-            default:
-                parentLanguage = childLanguage
-                break
+        const runtimeLanguage: codewhispererClient.ProgrammingLanguage = {
+            languageName: this.toRuntimeLanguage(
+                request.fileContext.programmingLanguage.languageName as CodewhispererLanguage
```

Comment:
I think it's confusing that this method is called to mapToRuntimeLanguage when it's in fact mapping to telemetry language, should we rename? and if we are using `normalizedLanguage` as the type for sending to service, let's keep it consistent, and toRuntimeLanguage is essentially toNormalizedLanguage?

Reply:
yes I am open to that, we can name it `normalizeLanguage()`

Reply:
> it's in fact mapping to telemetry language
that's why here the return type is `NormalizedLanguage`

Reply:
still the best solution and long-term solution is to introduce another language interface which should be used universally in our VSC plugin code base instead of relying on telemetry.gen type. However with that approach, the surface of changes might be huge and it would be better off refactoring component by component

Reply:
I see now it is mapping to NormalizedLanguage not the telemetry one I mis-read.If we can just use one of `runtimeLanguage` or `normalizedLanguage` to reduce one additional name that would be great